blinky/ui/src/main/java/no/nordicsemi/android/blinky/control/BlinkyScreen.kt

```diff
@@ -0,0 +1,111 @@
+package no.nordicsemi.android.blinky.control
+
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.Button
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import no.nordicsemi.android.blinky.control.view.ButtonControlView
+import no.nordicsemi.android.blinky.control.view.LedControlView
+import no.nordicsemi.android.blinky.control.viewmodel.BlinkyViewModel
+import no.nordicsemi.android.blinky.spec.Blinky
+import no.nordicsemi.android.common.logger.view.LoggerAppBarIcon
+import no.nordicsemi.android.common.permission.RequireBluetooth
+import no.nordicsemi.android.common.theme.view.NordicAppBar
+import no.nordicsemi.android.common.ui.scanner.view.DeviceConnectingView
+import no.nordicsemi.android.common.ui.scanner.view.DeviceDisconnectedView
+import no.nordicsemi.android.common.ui.scanner.view.Reason
+
+@Composable
+fun BlinkyScreen(
+    onNavigateUp: () -> Unit,
+) {
+    val viewModel: BlinkyViewModel = hiltViewModel()
+    val state by viewModel.state.collectAsState()
```

Comment:
I would recommend using `collectasstatewithlifecycle()` which is lifecycle aware. This would require the following dependency `lifecycle-viewmodel-compose` version [2.6.0-alpha02](https://developer.android.com/jetpack/androidx/releases/lifecycle#lifecycle-viewmodel-compose-2.6.0-alpha02).